src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -590,15 +590,37 @@ func newPromTSIter(timeseries []prompb.TimeSeries, tagOpts models.TagOptions) (*
 
 type promTSIter struct {
 	idx        int
+	err        error
 	attributes []ts.SeriesAttributes
 	tags       []models.Tags
 	datapoints []ts.Datapoints
 	metadatas  []ts.Metadata
+	annotation []byte
 }
 
 func (i *promTSIter) Next() bool {
 	i.idx++
-	return i.idx < len(i.tags)
+	if i.idx >= len(i.tags) {
+		return false
+	}
+
+	annotationPayload, err := storage.SeriesAttributesToAnnotationPayload(i.attributes[i.idx])
+	if err != nil {
+		i.err = err
+		return false
+	}
+
+	i.annotation, err = annotationPayload.Marshal()
+	if err != nil {
+		i.err = err
+		return false
+	}
+
+	if len(i.annotation) == 0 {
+		i.annotation = nil
+	}
```

Comment:
I'm not 100% sure. I believe I wanted to preserve the exact behaviour (in case the input stays the same: no types provided by the ingestion pipeline - the output also stays the same: nil instead of empty slice).

Reply:
:+1: they should be functionally identical but guessing the tests will be checking for nil explicitly?

Reply:
Yes, there is a test that fails with `Error:		Expected nil, but got: []byte{}` if I comment this out.